rotkehlchen/exchanges/woo.py

```diff
@@ -0,0 +1,431 @@
+                response=response,
+                case='balances',
+            )
+            return result, msg
+        try:
+            response_dict = jsonloads_dict(response.text)
+        except JSONDecodeError as e:
+            msg = f'Woo returned invalid JSON response: {response.text}.'
+            log.error(msg)
+            raise RemoteError(msg) from e
+        account_balances = self._deserialize_accounts_balances(response_dict=response_dict)
+        return account_balances, ''
+
+    def _deserialize_accounts_balances(
+            self,
+            response_dict: dict[str, Any],
+    ) -> dict[AssetWithOracles, Balance]:
+        try:
+            balances = response_dict['data']['holding']
+        except KeyError as e:
+            msg = 'Woo balances JSON response is missing data key'
+            log.error(msg, response_dict)
+            raise RemoteError(msg) from e
+
+        assets_balance: defaultdict[AssetWithOracles, Balance] = defaultdict(Balance)
+        for entry in balances:
+            symbol = entry.get('token')
+            try:
+                amount = deserialize_asset_amount(entry.get('holding') + entry.get('staked'))
+                if amount == ZERO:
+                    continue
+                asset = asset_from_woo(symbol)
+            except DeserializationError as e:
+                log.error(
+                    'Error processing a Woo balance.',
+                    entry=entry,
+                    error=str(e),
+                )
+                self.msg_aggregator.add_error(
+                    'Failed to deserialize a Woo balance. '
+                    'Check logs for details. Ignoring it.',
+                )
+                continue
+            except (UnknownAsset, UnsupportedAsset) as e:
+                log.error(str(e))
+                asset_tag = 'unknown' if isinstance(e, UnknownAsset) else 'unsupported'
+                self.msg_aggregator.add_warning(
+                    f'Found {asset_tag} Woo asset {e.identifier}. Ignoring its balance query.',
+                )
+                continue
+            try:
+                usd_price = Inquirer().find_usd_price(asset=asset)
+            except RemoteError as e:
+                log.error(str(e))
+                self.msg_aggregator.add_error(
+                    f'Error processing Woo balance result due to inability to '
+                    f'query USD price: {e!s}. Skipping balance entry.',
+                )
+                continue
+
+            assets_balance[asset] += Balance(
+                amount=amount,
+                usd_value=amount * usd_price,
+            )
+
+        return dict(assets_balance)
+
+    def _deserialize_trade(
+            self,
+            trade: dict[str, Any],
+    ) -> Trade:
+        """Deserialize a Woo trade returned from the API to History Event"""
+        symbol = trade['symbol']
+        _, base_asset_symbol, quote_asset_symbol = symbol.split('_')
```

Comment:
if for some reason they change the key for their trades this can raise unpack error. Protect the app against it